### DISCUSSION TITLE
Clear state in AsyncNotifier

### DISCUSSION THREAD

**Discussion by aidanmack** [rrousselGit/riverpod#2703] [Q&A] [answered]
Hello,
Im using the new AsyncNotifier.
Im trying to clear/reset the state of the provider before I  watch it, but it still emits the previous state first.

I've tried calling Invalidate before the widget where the watch is created.
`ref.invalidate(provider)`

I can see the build function in the provider gets invoked, but instead of emitting the `loading` state it emits the last State object in this case type `DetailedStation`.
It then makes the async call (does not emit the loading state while doing so) and then emits a new `DetailedStation`

This is my provider...

```
@riverpod
class AutoDisposeAsyncFullDetails extends _$AutoDisposeAsyncFullDetails {
  late ApiService apiService;

  @override
  FutureOr<DetailedStation> build() async {
    print("BUILDING");
    apiService = ref.watch(ApiCallsProvider);

      return await apiService.getStationsByIdNoTry("0", '1').then(
          (value) => Mapping.mapDomainStationToDetailedStation(
        domainStationDetail: value.first,
        stationType: "foo",
      ),
    );
  }
}
```

Any help/thoughts?

Thanks
Aidan

**Comment by AhmedLSayed9** [comment, marked as answer]
Loading state is skipped by default when using `ref.invalidate/ref.refresh` (as it's common to be used with RefreshIndicator).
To disable it and show loading, set `skipLoadingOnRefresh: false` at `.when`

**Reply by aidanmack** [reply]
Thats great - worked a treat, thankyou
